common/djangoapps/database_fixups/migrations/0002_remove_progress_extension.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+"""
+The `progress-edx-platform-extensions` has been deprecated in favor of `edx-completion`.
+
+The requirement was removed in the commit linked as (1) below. However its migration (2) had not been reverted.
+That migration used `auth_user.id` as the foreign key in its models (3), but Django does not resolve this constraint
+between existing tables anymore, because the model has been removed.
+Therefore we need to drop the tables related to deprecated application in order to be able to remove users properly.
+
+(1) https://github.com/edx-solutions/edx-platform/commit/59bf3efe71533de53b60bd979517e889d18a96bb
+(2) https://github.com/edx-solutions/progress-edx-platform-extensions/blob/master/progress/migrations/0001_initial.py
+(3) https://github.com/edx-solutions/progress-edx-platform-extensions/blob/master/progress/models.py
+"""
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('database_fixups', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.RunSQL('DROP TABLE IF EXISTS '
+                          'progress_coursemodulecompletion, progress_studentprogress, progress_studentprogresshistory')
+    ]
```

Comment:
I think this is probably fair to do, but I will leave it to edX to decide. I think this could be a pretty long operation. Although, my understanding is, so will dropping just the constraint.

Reply:
Who should we ping for this?

Reply:
@ihtram Could you review this change?

Reply:
@Agrendalath @xitij2000 `rebase-ironwood` is no more in use so for now we are merging everything in `integration` branch. Please change the target branch to integration. 
Secondly, two of the above tables we are going to drop have more than 60 Million records that could take a long time to complete the process. Although there would be table locking during this migration but in my opinion if we are not using theses tables, locking would not be a trouble for us. What do you think?

Reply:
@ihtram, thanks, I changed the target.
>  Secondly, two of the above tables we are going to drop have more than 60 Million records each, that could take a long time to complete the process. Although there would be table locking during this migration but in my opinion if we are not using theses tables, locking would not be a trouble for us. What do you think?

Right, we are not using them anymore, so it should be fine.
Any thoughts, @xitij2000?

Reply:
@Agrendalath I would recommend to drop these tables in two steps. Create a migration to drop first two tables, ship it to prod and then drop the last one in another deployment. We are not sure how much time it will take to drop two tables with 60million records each.

Reply:
@ihtram, sure we can do this. We will split this into two separate PRs tomorrow.

Reply:
@ihtram, marking as ready for review again, per comments on the upstream ticket.